modules/reindex/src/internalClusterTest/java/org/elasticsearch/index/reindex/ReindexPluginMetricsIT.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.reindex;
+
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.plugins.PluginsService;
+import org.elasticsearch.reindex.BulkIndexByScrollResponseMatcher;
+import org.elasticsearch.reindex.ReindexPlugin;
+import org.elasticsearch.search.sort.SortOrder;
+import org.elasticsearch.telemetry.Measurement;
+import org.elasticsearch.telemetry.TestTelemetryPlugin;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static org.elasticsearch.index.query.QueryBuilders.termQuery;
+import static org.elasticsearch.reindex.DeleteByQueryMetrics.DELETE_BY_QUERY_TIME_HISTOGRAM;
+import static org.elasticsearch.reindex.ReindexMetrics.REINDEX_TIME_HISTOGRAM;
+import static org.elasticsearch.reindex.UpdateByQueryMetrics.UPDATE_BY_QUERY_TIME_HISTOGRAM;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 0, numClientNodes = 0, scope = ESIntegTestCase.Scope.TEST)
+public class ReindexPluginMetricsIT extends ESIntegTestCase {
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(ReindexPlugin.class, TestTelemetryPlugin.class);
+    }
+
+    protected ReindexRequestBuilder reindex() {
+        return new ReindexRequestBuilder(client());
+    }
+
+    protected UpdateByQueryRequestBuilder updateByQuery() {
+        return new UpdateByQueryRequestBuilder(client());
+    }
+
+    protected DeleteByQueryRequestBuilder deleteByQuery() {
+        return new DeleteByQueryRequestBuilder(client());
+    }
+
+    public static BulkIndexByScrollResponseMatcher matcher() {
+        return new BulkIndexByScrollResponseMatcher();
+    }
+
+    public void testReindexMetrics() throws Exception {
+        internalCluster().startNode();
+
+        indexRandom(
+            true,
+            prepareIndex("source").setId("1").setSource("foo", "a"),
+            prepareIndex("source").setId("2").setSource("foo", "a"),
+            prepareIndex("source").setId("3").setSource("foo", "b"),
+            prepareIndex("source").setId("4").setSource("foo", "c")
+        );
+        assertHitCount(prepareSearch("source").setSize(0), 4);
+
+        final String dataNodeName = internalCluster().getRandomNodeName();
```

Comment:
since this test uses a single node started with `internalCluster().startNode();` can we get the name from that node instead of using `getRandomNode` that way it's clearer that we'll be fetching the metrics from that node.

Reply:
Yes! Let me make the change.